packages/react/src/tokens/themes/theme.types.ts

```diff
@@ -1,12 +1,59 @@
 import { Colors } from '../palettes/palette';
-import { BreakpointsType, Typography } from './theme.constants';
+import { BreakpointsType, FontWeight, FontFamilyType } from './theme.constants';
 
 export interface Theme {
-  fontSize?: number;
-  fontFamily?: string;
-  spaceUnit?: object;
-  fontWeight?: number;
-  colors?: Colors;
-  typography?: Typography;
-  breakpoints?: BreakpointsType;
+  name: string;
+  fontSize: number;
+  fontFamily: FontFamilyType<'display' | 'text'>;
+  spaceUnit: Record<string, string>;
+  fontWeight: typeof FontWeight;
+  colors: Colors;
+  typography: Typography;
+  breakpoints: BreakpointsType;
 }
+
+interface TypographyStyle {
+  fontSize?: string;
+  fontWeight?: FontWeight;
+  lineHeight?: string;
+}
+
+interface TypographyBreakpoints {
+  mobile?: TypographyStyle;
+  tablet?: TypographyStyle;
+  desktop?: TypographyStyle;
+  desktopLarge?: TypographyStyle;
+}
+
+export interface TypographyHeader {
```

Comment:
Makes sense for headings and other type of same structures?

```suggestion
export enum Headings {
  H1 = 'h1',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  H6 = 'h6'
}

type TypographyHeader = {
  [heading in Headings]: {
    [breakpoint in Breakpoints]: TypographyStyle
  };
}
```